### PR TITLE
Updating error handling in runs_start_blocking

### PR DESCRIPTION
**Pull request by macrogers**

### CODE REVIEW

File: domino/domino.py [hunk 1]
```diff
@@ -136,16 +136,18 @@ def runs_start_blocking(self, command, isDirect=False, commitId=None,
...
         run_id = run_response['runId']
-
+        
         poll_start = time.time()
         current_retry_count = 0
         while True:
             try:
                 run_info = self.get_run_info(run_id)
+                if run_info is None:
+                    raise RunNotFoundException(f"Tried to access nonexistent run id {run_id}")
```
Comment (abhijeet2096) [inline domino/domino.py:146]: Please add an import statement at the top
```
from .exceptions import RunNotFoundException
```

File: domino/domino.py [hunk 2]
```diff
@@ -136,16 +136,18 @@ def runs_start_blocking(self, command, isDirect=False, commitId=None,
...
         poll_start = time.time()
         current_retry_count = 0
         while True:
             try:
                 run_info = self.get_run_info(run_id)
+                if run_info is None:
+                    raise RunNotFoundException(f"Tried to access nonexistent run id {run_id}")           
                 current_retry_count = 0
-            except requests.exceptions.RequestException as e:
+            except (Exception) as e:
```
Comment (abhijeet2096) [inline domino/domino.py:148]: Add specific exceptions here
```
except (requests.exceptions.RequestException, RunNotFoundException) as e: 
```

**Comment by wseaton** [comment]
While we are at it, can the `Exception` thrown on run failure also get it's own exception class? Maybe something like `RunFailedException` (even `RuntimeError` could work)?

It would make catching the log and cleanly formatting it easier on run failure, which is something the `DominoOperator` has to do.

File: domino/domino.py (hunk 1 above)
Comment (macrogers) [reply domino/domino.py:148]: Done

**Comment by macrogers** [comment]
@abhijeet2096 updates added, let me know what you think. 

@wseaton I went ahead and added a RunFailedException as well.

File: domino/domino.py [hunk 3]
```diff
@@ -136,16 +138,18 @@ def runs_start_blocking(self, command, isDirect=False, commitId=None,
         run_response = self.runs_start(command, isDirect, commitId, title,
                                        tier, publishApiEndpoint)
         run_id = run_response['runId']
-
+
```
Comment (wseaton) [inline domino/domino.py:141]: trailing whitespace

**Review by abhijeet2096 — APPROVED** [review]
Looks good. Please remove the trailing space and merge it. 👍

File: domino/domino.py (hunk 3 above)
Comment (macrogers) [reply domino/domino.py]: done

**Pull request merged**